include/seqan3/io/detail/take_until_view.hpp

```diff
@@ -0,0 +1,650 @@
+ * if the end of the underlying range is reached before their own termination criterium is met. This is useful
+ * if you want a "strict" evaluation of the functor.
+ *
+ * Consuming: `seqan3::views::take_until_and_consume` and `seqan3::views::take_until_or_throw_and_consume` behave the same
+ * as their non-consuming counter-parts if the underlying range models at least `std::forward_range`.
+ * If, however, the underlying range is a pure `std::input_range`, the view will keep moving the underlying
+ * iterator forward as long as the termination criterium holds and the underlying range is not at end.
+ * This is useful for string tokenisation among other things.
+ *
+ * ### Example
+ *
+ * \include test/snippet/io/detail/take_until_view.cpp
+ *
+ * \hideinitializer
+ *
+ * \deprecated Use std::views::take_while with std::not_fn.
+ */
+inline auto constexpr take_until = detail::take_until_fn<false, false>{};
+
+// ============================================================================
+//  views::take_until_or_throw (adaptor instance definition)
+// ============================================================================
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (**throws** if the end of the underlying range is reached).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup views
+ *
+ * \copydetails seqan3::views::take_until
+ * \hideinitializer
+ */
+inline auto constexpr take_until_or_throw = detail::take_until_fn<true, false>{};
+
+// ============================================================================
+//  views::take_until_and_consume (adaptor instance definition)
+// ============================================================================
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (or the end of the underlying range is reached; consumes end in single-pass ranges).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup views
+ *
+ * \copydetails seqan3::views::take_until
+ * \hideinitializer
+ * \deprecated A previous call `view | take_until_and_consume(is_space)` could be convert to
+ * `view | std::take_while(std::not_fn(is_space)) | std::drop_while(is_space)`
+ */
+inline auto constexpr take_until_and_consume = detail::take_until_fn<false, true>{};
+
+// ============================================================================
+//  views::take_until_or_throw_and_consume (adaptor instance definition)
+// ============================================================================
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (**throws** if the end of the underlying range is reached; consumes end in single-pass ranges).
```

Comment:
```suggestion
 *        (**throws** if the end of the underlying range is reached and consumes the end in single-pass ranges).
```

Reply:
still open?

Reply:
Sorry, I didn't notice that this text appears multiple times, only fixed on of them. Now they are all fixed! :+1: